### ISSUE TITLE
"[DEP0040] DeprecationWarning" when running `nodecg setup` on Node.js v22 LTS

### ISSUE THREAD

**Issue by zoton2** [nodecg/nodecg-cli#97]
(I think this also applies to v21 but I have only tested it with an LTS version of 22.)

When running the `nodecg setup` command on Node.js v22, this warning is printed in the console:
```
Downloading v2.3.0 from npm... (node:4920) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
```
I did some investigation and this is because of the `node-fetch` v2.x dependency, which uses the `whatwg-url` v5.x dependency, which references `punycode`. It's important to note this is the Node.js *internal* version of `punycode` and not the one from NPM. `whatwg-url` removed the need for the dependency in v9.x, and I don't think `node-fetch` v3.x uses the former anymore anyway, but it's not as simple as upgrading to v3.x as it's a an ESM only package, which I assume would need more changes in this library.

It's important to also note that this is currently only a warning, so it all still works, but obviously is something to be concerned about.

The current workaround from other people on other libraries with the same issue seems to be to just override `whatwg-url` with a newer version, similar to what we just did with another dependency in NodeCG itself (https://github.com/nodecg/nodecg/pull/760), like this (including an example where it specifically targets `node-fetch@2.x` so it doesn't unintentionally break another dependency, taken from [this comment](https://github.com/node-fetch/node-fetch/issues/1797#issuecomment-2283353733)):
```
"overrides": {
  "node-fetch@2.x": {
    "whatwg-url": "14.x"
  }
}
```
From skimming the release notes of `whatwg-url` I don't see a reason we can use v14.x (which is latest), as long as we're OK only supporting Node.js v18 and above (which is the same as NodeCG itself).

**Comment by zoton2** [comment]
Another quick note is I saw the same error when using `nodecg schema-types` as well, but I'm not sure what triggers it for that; maybe `json-schema-to-typescript` but didn't look too deeply into it.

**Comment by Hoishin** [comment]
fixed with v8.7.0
